src/cluster_legacy.c

```diff
@@ -2849,20 +2909,27 @@ int clusterProcessPacket(clusterLink *link) {
     clusterMsg *hdr = (clusterMsg *)link->rcvbuf;
     uint16_t type = ntohs(hdr->type);
     mstime_t now = mstime();
+    clusterNode *sender = getNodeFromLinkAndMsg(link, hdr);
+    /* Update the last time we saw any data from this node. We
+     * use this in order to avoid detecting a timeout from a node that
+     * is just sending a lot of data in the cluster bus, for instance
+     * because of Pub/Sub. */
+    if (sender) sender->data_received = now;
+
+    if (sender && (type == CLUSTERMSG_TYPE_PUBLISH || type == CLUSTERMSG_TYPE_PUBLISHSHARD) && nodeSupportsPubsubMsgHdr(sender)) {
+        return pubsubProcessLightPacket(link, type);
+    }
```

Comment:
question: what prevents us from moving the handling of old pubsub msg processing here as well?

Reply:
I though of it, but then did not want to change the flow and minimize code changes. I can try moving it here if thats what we would prefer.

Reply:
> I though of it, but then did not want to change the old pubsub msg processing behavior and minimize code changes. I can try moving it here if thats what we would prefer.

Yeah, let's try to simplify the code as much as possible. Most people will just read the new code, and we want as few branching paths as possible.